packages/types/src/srml/balances/definitions.ts

```diff
@@ -2,12 +2,13 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import U32 from '../primitive/U32';
-
-/**
- * @name ApprovalFlag
- * @description
- * Approval Flag, implemented as a [[U32]]
- */
-export default class ApprovalFlag extends U32 {
-}
+export default {
+  types: {
+    BalanceLock: {
+      id: 'LockIdentifier',
+      amount: 'Balance',
+      until: 'BlockNumber',
+      reasons: 'WithdrawReasons'
```

Comment:
WithdrawReasons is a set - we don't have a json-like definition for set (so that one cannot move)